File: desktopApp/core/src/main/java/si/um/feri/mori/raster/Garden.java
package si.um.feri.mori.raster;

import com.badlogic.gdx.utils.Array;

public class Garden {
    public String _id;
    public String name;
    public String owner;
    public int width;
    public int height;
    public Array<Element> elements;
    public double latitude;
    public double longitude;
    public String location;
    public String createdAt;
    public String updatedAt;
    public int __v;

    public static class Element {
        public int x;
        public int y;
        public String type;
        public String crop;
        public String plantedDate;
        public String wateredDate;
        public String _id;
    }
}

File: KotlinApp/src/main/kotlin/models/Garden.kt
package models

import kotlinx.serialization.Serializable

@Serializable
data class Tile(
    val x: Int,
    val y: Int,
    val type: String,
    val crop: String? = null,
    val plantedDate: String? = null,
    val wateredDate: String? = null,
    val _id: String? = null
)

@Serializable
data class Garden(
    val _id: String,
    val name: String,
    val owner: String,
    val width: Int,
    val height: Int,
    val location: String? = null,
    val latitude: Double? = null,
    val longitude: Double? = null,
    val elements: List<Tile> = emptyList(),
    val createdAt: String? = null,
    val updatedAt: String? = null
)